src/Store/DoctrineStore.php

```diff
@@ -48,6 +56,97 @@ public function connection(): Connection
+    {
+        $this->connection->transactional(
+            function (Connection $connection) use ($messages): void {
+                foreach ($messages as $message) {
+                    $event = $message->event();
+
+                    $data = $this->serializer->serialize($event);
+
+                    $connection->insert(
+                        self::OUTBOX_TABLE,
+                        [
+                            'aggregate' => $this->aggregateRootRegistry->aggregateName($message->aggregateClass()),
+                            'aggregate_id' => $message->aggregateId(),
+                            'playhead' => $message->playhead(),
+                            'event' => $data->name,
+                            'payload' => $data->payload,
+                            'recorded_on' => $message->recordedOn(),
+                        ],
+                        [
+                            'recorded_on' => Types::DATETIMETZ_IMMUTABLE,
+                        ]
+                    );
+                }
+            }
+        );
+    }
+
+    /**
+     * @return list<Message>
+     */
+    public function retrieveOutboxMessages(?int $limit = null): array
+    {
+        $sql = $this->connection->createQueryBuilder()
+            ->select('*')
+            ->from(self::OUTBOX_TABLE)
+            ->setMaxResults($limit)
+            ->getSQL();
+
+        /** @var list<array{aggregate: string, aggregate_id: string, playhead: string|int, event: string, payload: string, recorded_on: string}> $result */
+        $result = $this->connection->fetchAllAssociative($sql);
+        $platform = $this->connection->getDatabasePlatform();
+
+        return array_map(
+            function (array $data) use ($platform) {
+                return new Message(
+                    $this->aggregateRootRegistry->aggregateClass($data['aggregate']),
+                    $data['aggregate_id'],
+                    self::normalizePlayhead($data['playhead'], $platform),
+                    $this->serializer->deserialize(new SerializedEvent($data['event'], $data['payload'])),
+                    self::normalizeRecordedOn($data['recorded_on'], $platform)
+                );
+            },
+            $result
+        );
+    }
+
+    public function markOutboxMessageConsumed(Message ...$messages): void
+    {
+        $this->connection->transactional(
+            function (Connection $connection) use ($messages): void {
+                foreach ($messages as $message) {
+                    $connection->delete(
+                        self::OUTBOX_TABLE,
+                        [
+                            'aggregate' => $this->aggregateRootRegistry->aggregateName($message->aggregateClass()),
+                            'aggregate_id' => $message->aggregateId(),
+                            'playhead' => $message->playhead(),
+                        ]
+                    );
+                }
+            }
+        );
+    }
```

Comment:
This is not really "marking" the message as consumed, but more a "delete consumed messages". But i think, this should be fine.

Reply:
From the API side you just want to say that the message was processed. But what you then do with the info in the implementation doesn't really matter.

Maybe we should remove the `mark` prefix?

`markOutboxMessageConsumed` => `outboxMessageConsumed`

Reply:
Tbh, i dont think so. In our usecase marking is deleting. Maybe someone else implements it different and really only marks the message since they dont want do delete it and loose the data. So i think this is fine as it is.